operator/pkg/name/name.go

```diff
@@ -80,37 +77,45 @@ var (
 		CitadelComponentName,
 		CNIComponentName,
 	}
-	DeprecatedNames = []ComponentName{
-		InjectorComponentName,
-		SidecarInjectorComponentName,
-		NodeAgentComponentName,
-	}
-	AllLegacyAddonComponentNames = []ComponentName{
-		PrometheusComponentName,
-		KialiComponentName,
-		GrafanaComponentName,
-		TracingComponentName,
-		CoreDNSComponentName,
-	}
-	allComponentNamesMap         = make(map[ComponentName]bool)
-	deprecatedComponentNamesMap  = make(map[ComponentName]bool)
-	LegacyAddonComponentNamesMap = make(map[ComponentName]bool)
-	LegacyAddonComponentPathMap  = make(map[string]string)
+	allComponentNamesMap = make(map[ComponentName]bool)
+	// DeprecatedComponentNamesMap defines the istio component name used in old versions.
```

Comment:
this explanation is not really sufficient i.e. to someone without a history of how this came about would be scratching their heads.

Reply:
added more explanation